ref/Microsoft.Build/net/Microsoft.Build.cs

```diff
@@ -1495,6 +1496,19 @@ public partial class ProxyTargets
 }
 namespace Microsoft.Build.FileSystem
 {
+    public delegate bool FindPredicate(ref System.ReadOnlySpan<char> fileName);
+    public delegate TResult FindTransform<TResult>(ref System.ReadOnlySpan<char> fileName);
+    public partial interface IDirectoryCache
```

Comment:
Wouldn't it be better to augment MSBuildFileSystemBase with these new methods instead of creating a new interface?. With a new interface, the APIs will get harded to use. Users will have to think whether to implement one or the other, or both, Maybe a new generic MSBuildFileSystemBase<T> that extends MSBuildFileSystemBase.

Reply:
I pushed an update where the interface is now provided via `Project` and not `EvaluationContext` so having a separate one should be easier to justify. I would argue that having multiple methods providing the same functionality in `MSBuildFileSystemBase` would also be confusing for users because it wouldn't be clear which one is used when.